fraos/src/seqno.rs

```diff
@@ -114,12 +142,47 @@ impl SeqNoIndex {
     pub(crate) fn mmaps_count(&self) -> Result<usize, FraosError> {
         self.inner.mmaps_count()
     }
+
+    pub(crate) fn find_actual_end(&self) -> Result<usize, FraosError> {
+        let mut start = 0;
+        let len = self.len();
+        let mut end = self.len();
+
+        // empty index was created or index is empty
+        if self.get_length_at(start)? == 0 || end == 0 {
+            return Ok(0);
+        }
+
+        // all elements are non-zero
+        if self.get_length_at(end.saturating_sub(1))? != 0 {
+            return Ok(end);
+        }
+
+        while start < len.saturating_sub(1) {
```

Comment:
maybe we could have a comment explaining that here we should have at least 2 elements because of the previous early return?